src/sql/workbench/contrib/objectExplorer/browser/serverTreeView.ts

```diff
@@ -527,9 +527,12 @@ export class ServerTreeView extends Disposable implements IServerTreeView {
 				await this._connectionManagementService.deleteConnection(profile);
 			}
 			const connectionProfile = this.getConnectionInTreeInput(profile.id);
-
+			// If the connection is not found in the tree, it means it was already deleted from the tree and we don't need to disconnect it
+			if (!connectionProfile) {
+				return;
+			}
 			// For the connection profile, we need to clear the password from the last session if the user doesn't want to save it
-			if (!connectionProfile.savePassword) {
+			if (!connectionProfile?.savePassword) {
```

Comment:
nit - connectionProfile should always be defined here with what you've added above
```suggestion
			if (!connectionProfile.savePassword) {
```